### DISCUSSION TITLE
Support for flatgeobuf format

### DISCUSSION THREAD

**Discussion by jaredlander** [ropensci/targets#599] [Ideas]
## Prework

* [X] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [X] If there is [already a relevant issue](https://github.com/ropensci/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [X] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.
* [X] Format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Proposal

When a target results in a large object the standard 'rds' format can be slow. For `data.frame`s and alike alternatives like 'fst' and 'parquet' are pretty great. However, `sf` objects don't work well with that format (it's surprising that 'parquet' doesn't play nicely). Even 'qs' gives me trouble.

So I'd love to see a 'flatgeobuf' option. The `{sf}` package supports writing to flatgeobuf by calling `sf::sf_write(x, 'file_name.fgb')` if `gdal` > 3.1 is [installed](https://github.com/r-spatial/sf/issues/1402) (I can't get that installed on my machine for some reason so I wasn't able to test this myself).

My thinking would be that setting `format='flatgeobuf'` would write the `sf` object to this type of file quickly.

```R
list(
    tar_target(
        one_point,
        tibble::tibble(ID=1:2, X=c(54.24, 52.34), Y=c(34.17, 33.31)) %>% 
            sf::st_as_sf(coords=c('X', 'Y')),
        packages=c('magrittr'),
        format='flatgeobuf'
    )
)
```

I tried doing this myself for a PR my copying the [parquet format](https://github.com/ropensci/targets/blob/main/R/class_parquet.R), but the wrong version of gdal is stopping me (and I know I need to fix that before I could even use this feature).

**Comment by wlandau** [comment]
I would still prefer other solutions than a built-in format for this. I try to keep formats general: data frames, popular machine learning objects, and R objects in general. `sf` is popular, but it is still domain-specific enough to give me pause.

I also feel like `arrow` is really close to being able to just save `sf` objects as feather or parquet files. What have the maintainers of those packages said about this? If either `sf` or `arrow` could bridge the gap, then it would benefit more R users than just `targets` users.

**Reply by wlandau** [reply]
One extra format does not seem like much at first, but I have started to add `"aws_*"` versions of all formats, and I am considering Google Cloud and Azure. Those are the major players in the cloud right now, but even more could emerge. Local formats X cloud providers is already enough combinatorics to be dangerous.
